### PR TITLE
JBPM-7031: Stunner - Disable doing operation in view mode

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/JBPM-7031

@romartin I tried endlessly to only have one "open" method and cast the ```SessionPresenter``` but could not find a way to do it; so had to (unhappily) settle for what you see in the PR (two ```SessionPresenter```'s, one for "full" and another for "read-only").... if you can explain how to cast I'll more than happily update to use a single ```openSession(..)``` method!

### CODE REVIEW

**Comment by manstis** [comment]
@romartin I'll hold off on the 7.7.x PR until everyone is happy with the solution (re: comment above).

File: kie-wb-common-stunner/kie-wb-common-stunner-extensions/kie-wb-common-stunner-project/kie-wb-common-stunner-project-client/src/test/java/org/kie/workbench/common/stunner/project/client/editor/AbstractProjectDiagramEditorTest.java [hunk 1]
```diff
@@ -415,25 +466,76 @@ public void testOpen() {
...
         verify(view).setWidget(eq(sessionPresenterView));
-        verify(sessionPresenter).withToolbar(eq(false));
-        verify(sessionPresenter).withPalette(eq(true));
-        verify(sessionPresenter).open(eq(diagram),
-                                      eq(clientFullSession),
-                                      clientSessionPresenterCallbackCaptor.capture());
+        verify(readOnlySessionPresenter).withToolbar(eq(false));
+        verify(readOnlySessionPresenter).withPalette(eq(false));
+        verify(readOnlySessionPresenter).open(eq(diagram),
+                                              eq(clientFullSession),
```
Comment (jomarko) [inline kie-wb-common-stunner/kie-wb-common-stunner-extensions/kie-wb-common-stunner-project/kie-wb-common-stunner-project-client/src/test/java/org/kie/workbench/common/stunner/project/client/editor/AbstractProjectDiagramEditorTest.java]: I would expect `clientReadOnlySession` or am I wrong?

**Comment by jomarko** [comment]
@manstis some DMN tests probably need your love if you have a look on CI results.

File: kie-wb-common-stunner/kie-wb-common-stunner-extensions/kie-wb-common-stunner-project/kie-wb-common-stunner-project-client/src/test/java/org/kie/workbench/common/stunner/project/client/editor/AbstractProjectDiagramEditorTest.java (hunk 1 above)
Comment (manstis) [reply kie-wb-common-stunner/kie-wb-common-stunner-extensions/kie-wb-common-stunner-project/kie-wb-common-stunner-project-client/src/test/java/org/kie/workbench/common/stunner/project/client/editor/AbstractProjectDiagramEditorTest.java]: Good spot. Will fix along with the DMN tests...

**Comment by manstis** [comment]
@jomarko Updated with new commit.

**Comment by manstis** [comment]
@jomarko Hello, I looked into what you report and it is (a) not introduced by this PR (the same happens with {{master}}) and (b) is by design and consistent with how failed lock attempts work for other editors in the workbench (e.g. DRL editor). Following a failure to acquire a lock AppFormer notifies the User and reloads the editor - not in "view/read-only" mode, but normal "editing" mode.

My suggestion would be, to test this PR, create multiple versions of a process file and select older versions (from either the dropdown or Overview tab) to "activate" view mode.

**Comment by manstis** [comment]
@jomarko @romartin Ready for review :-)

**Comment by romartin** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
Hey @manstis ! sorry for late responses.. had been a crazy week... 

Let you know that I've changed this part about the commands a bit in my local changes about performance, so I'll rebase once this is pushed and probably will have to change a bit.

On the other hand, just trying to see the scope for this task, tbh not sure the actual state of this, so just asking.. The goal is to disable the edition when moving to older versions, so you added here the logic for the editor buttons for this, right? Anyway when moving to view mode, Stunner should be presented using the "read-only" mode, is that already working? What's doing now?
Thx in advance for the info!

**Comment by manstis** [comment]
@romartin This PR is the complete solution. Use of Read Only sessions when applicable, disposal of sessions (when switching versions) and all required menu bar enable/disable function etc.

**Comment by manstis** [comment]
Jenkins please execute full downstream build.

**Comment by manstis** [comment]
Jenkins please execute full downstream build.

**Comment by manstis** [comment]
Jenkins please execute full downstream build.

**Comment by manstis** [comment]
Jenkins please execute full downstream build.

**Comment by manstis** [comment]
Jenkins please execute full downstream build.

**Comment by manstis** [comment]
Jenkins please execute full downstream build.

**Comment by manstis** [comment]
@romartin @jomarko Did you want to wait for a "full downstream" build before merge?

It seems to keep failing on the ```kie-wb-distribution``` Selenium/REST tests (time out deploying).

**Comment by jomarko** [comment]
green from me

**Pull request merged**